xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -117,32 +118,56 @@ void CPVRClients::UpdateAddons(const std::string& changedAddonId /*= ""*/)
       AddonInfoPtr addon = addonWithStatus.first;
       bool bEnabled = addonWithStatus.second;
 
-      if (bEnabled && (!IsKnownClient(addon->ID()) || !IsCreatedClient(addon->ID())))
+      std::vector<uint32_t> instanceIds;
+      if (addon->SupportInstanceSettings())
       {
-        int iClientId = ClientIdFromAddonId(addon->ID());
+        instanceIds = addon->GetKnownInstanceIds();
 
-        std::shared_ptr<CPVRClient> client;
-        if (IsKnownClient(addon->ID()))
+        // Insert the on function given instance id, needed after delete of instance and
+        // then no more available in GetKnownInstanceIds.
+        if (changedInstance > KODI::ADDONS::ADDON_SETTINGS_ID && changedAddonId == addon->ID() &&
```

Comment:
This also reads wrong/hacky. Again, waht have settings to do with addon creation? Do wen need another constant (with the same numerical value as ADDON_SETTINGS_ID) to denote sinle instance addon id? => like "ADDON_SINGLETON_INSTANCE_ID" ?

Reply:
Overseen, becomes changed to `ADDON_SINGLETON_INSTANCE_ID`